src/github.ts

```diff
@@ -0,0 +1,140 @@
+}
+
+/**
+ * Request reviews on the current context's PR from the specified reviewers.
+ * @param reviewers The list of reviewers to request a review.
+ */
+export async function requestReviewers(reviewers: string[]): Promise<void> {
+  const pr = getPr().number;
+  core.info(`Requesting Reviewers for PR #${pr} to: ${reviewers.join(',')}`);
+
+  const response = await client.rest.pulls.requestReviewers({
+    owner: github.context.repo.owner,
+    repo: github.context.repo.repo,
+    pull_number: pr,
+    reviewers: reviewers,
+  });
+  coreDebugJson(response, `requestReviewers(${pr}, [${reviewers.join(',')}]) > response`);
+}
+
+/**
+ *
+ * Assign the current context's PR to the specified assignees.
+ * @param assignees The list of assignees to assign the PR.
+ */
+export async function setAssignees(assignees: string[]): Promise<void> {
+  const pr = getPr().number;
+  core.info(`Setting Assignees for PR #${pr} to: ${assignees.join(',')}`);
+
+  const response = await client.rest.issues.addAssignees({
+    owner: github.context.repo.owner,
+    repo: github.context.repo.repo,
+    issue_number: pr,
+    assignees: assignees,
+  });
+  coreDebugJson(response, `setAssignees(${pr}, [${assignees.join(',')}]) > response`);
+}
+
+export type CheckRun = components['schemas']['check-run'];
+
+/**
+ * Get the checkruns for the current context's workflow_run for the head SHA.
+ */
+export async function getCheckRuns(): Promise<CheckRun[]> {
+  const ref = github.context.payload.workflow_run.head_sha;
+  const { owner, repo } = github.context.repo;
+
+  core.info(`Retrieving check runs for ${owner}/${repo}@${ref}...`);
+  const checkRuns: CheckRun[] = (
+    await client.rest.checks.listForRef({
+      owner,
+      repo,
+      ref,
+    })
+  ).data.check_runs;
+
+  core.info(`Retrieved ${checkRuns.length} check runs.`);
+  coreDebugJson(checkRuns, 'getCheckRuns() > checkRuns');
+
+  return checkRuns;
+}
+
+/**
+ * Get the required checks for the base ref.
+ */
+export async function getRequiredCheckNames(): Promise<string[] | undefined> {
+  const baseRef = getPr()?.base.ref;
+  core.info(`Base Ref: ${baseRef}`);
+  if (baseRef === undefined) {
+    core.error(`Error getting base ref for PR #${getPr()?.number}.`);
+    return undefined;
+  }
+  const { owner, repo } = github.context.repo;
+
+  core.info(`Retrieving branch protection information for ${owner}/${repo}@${baseRef}...`);
+  const result: components['schemas']['status-check-policy'] = (
+    await client.rest.repos.getStatusChecksProtection({
+      owner,
+      repo,
+      branch: baseRef,
+    })
+  ).data;
+
+  coreDebugJson(result, 'getRequiredChecks() > result');
+
+  return result?.contexts;
+}
+
+/**
+ * Get user inputs as an array (expects the user input to be CSV)
+ * @param name The name of the user input
+ * @param defaultVal The default value
+ */
+export function getInputArray(name: string, defaultVal: string[]): string[] {
+  return (
+    core
+      .getInput(name)
```

Comment:
What happens if this is undefined?

Reply:
This should never happen; method docstring states `Returns an empty string if the value is not defined.` If it *does* start returning undefined, this will raise an exception.